src/test/java/linked/list/single/SingleLinkedListTest.java

```diff
@@ -0,0 +1,319 @@
+package linked.list.single;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.NoSuchElementException;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.params.provider.Arguments.of;
+
+/**
+ *
+ * @author Dmitry Pavlov
+ * @since 19.04.2020
+ */
+class SingleLinkedListTest {
+
+    @Test
+    public void shouldCreateEmptyList() {
+        // when
+        var list = new SingleLinkedList<Integer>();
+        // then
+        assertEquals(list.getSize(), 0);
```

Comment:
shouldCreateEmptyList? - it's enough to check not null I think

Reply:
Checking size is not only about  nullability but initialisation of the `size` value.